code/modules/reagents/chemistry/equilibrium.dm

```diff
@@ -0,0 +1,226 @@
+/*
+An object/datum to contain the vars for each of the reactions currently ongoing in a holder/reagents datum
+This way all information is kept within one accessable object
+equilibrium is a unique name as reaction is already too close to chemical_reaction
+This is set up this way to reduce holder.dm bloat as well as reduce confusing list overhead
+The crux of the fermimechanics are handled here
+Instant reactions AREN'T handled here. See holder.dm
+*/
+/datum/equilibrium
+	var/datum/chemical_reaction/reaction //The chemical reaction that is presently being processed
+	var/datum/reagents/holder //The location the processing is taking place
+	var/multiplier = INFINITY
+	var/productRatio = 0
+	var/targetVol = INFINITY//The target volume the reaction is headed towards.
+	var/reactedVol = 0 //How much of the reaction has been made so far. Mostly used for subprocs
+	var/toDelete = FALSE //If we're done with this reaction so that holder can clear it
+
+/datum/equilibrium/New(datum/chemical_reaction/Cr, datum/reagents/R)
+	reaction = Cr
+	holder = R
+	if(!check_inital_conditions()) //If we're outside of the scope of the reaction vars
+		toDelete = TRUE
+		return
+	if(!calculate_yield())
+		toDelete = TRUE
+		return
+	reaction.on_reaction(holder, multiplier) 
+	SSblackbox.record_feedback("tally", "chemical_reaction", 1, "[reaction.type] attempts")
+
+//Check to make sure our input vars are sensible - truncated version of check_conditions (as the setup in holder.dm checks for that already)
+/datum/equilibrium/proc/check_inital_conditions()
+	if(holder.chem_temp > reaction.overheat_temp)//This is here so grenades can be made
+		SSblackbox.record_feedback("tally", "chemical_reaction", 1, "[reaction.type] overheats")
+		reaction.overheated(holder, src)//Though the proc will likely have to be created to be an explosion.
+
+	if(!reaction.is_cold_recipe)
+		if(holder.chem_temp < reaction.required_temp) //This check is done before in holder, BUT this is here to ensure if it dips under it'll stop
+			return FALSE //Not hot enough
+	else
+		if(holder.chem_temp > reaction.required_temp)
+			return FALSE //Not cold enough
+			
+	if(! ((holder.pH >= (reaction.optimal_pH_min - reaction.determin_pH_range)) && (holder.pH <= (reaction.optimal_pH_max + reaction.determin_pH_range)) ))//To prevent pointless reactions
+		return FALSE
+	return TRUE
+
+//Check to make sure our input vars are sensible - temp, pH, catalyst and explosion
+/datum/equilibrium/proc/check_conditions()
```

Comment:
Please give this proc a more informative name. It could mean anything without checking the documentation.

Reply:
Is check_reagent_properties() better?